### PR TITLE
Refine errors, re-define error codes

### PR DESCRIPTION
**Pull request by yah01**
/kind improvement
This is not finished, as we need to re-define error codes in proto

### CODE REVIEW

File: internal/util/errutil/errors.go [hunk 1]
```diff
@@ -18,10 +18,160 @@
...
+		err:  errors.New(msg),
+		code: code,
+	}
+}
+
+func (e milvusError) Error() string {
+	return e.err.Error()
+}
+
+func (e milvusError) Code() int32 {
```
Comment (jiaoew1991) [inline internal/util/errutil/errors.go:136]: does it need to be public? 
we already have a Code function below

**Comment by yah01** [comment]
rerun ut

**Review by congqixia — APPROVED** [review]
/lgtm

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**